src/Umbraco.Web/PublishedCache/XmlPublishedCache/XmlPublishedContent.cs

```diff
@@ -432,7 +433,20 @@ private void InitializeNode()
 		    }
 
             //lookup the content type and create the properties collection
-            contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+            try
+            {
+                contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+
+            }
+            catch (Exception e)
+            {
+                if (e.TargetSite.Name == "CreatePublishedContentType")
+                {
+                    content.Instance.RefreshContentFromDatabase();
+                    throw new Exception(string.Format("The content cache failed to find a content type with alias \"{0}\". This usually indicates that the content cache is corrupt; the content cache has been rebuilt in an attempt to self-fix the issue.", docTypeAlias));

```

Comment:
@zpqrtbnk  As above - please never throw generic exceptions,
 see https://umbraco.slack.com/archives/C0968J7PU/p1487752062130762 and https://umbraco.slack.com/archives/C0968J7PU/p1487750551111745 and https://msdn.microsoft.com/en-us/library/ms229007%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396